### PR TITLE
Pre-commit integration

### LINKED ISSUE #14: Configure pre-commit tool for this repo
> # Rules:
> 1. **DO NOT** commit to master. ever.
> 2. **ABSOLUTELY NO** force pushes allowed for public non-feature branches.
> 3. For every feature, create a new branch matching name `feature/<issue-number>-<feature-name>`.
> 4. Submit every feature by creating Pull Request @ github, after that wait for review, fix issues, repeat until approved. If needed rebase the branch on top of master and re-push force it.

  * [x] **I HAVE READ AND UNDERSTOOD THE RULES ABOVE**
------------

* [x] Install pre-commit locally as follows:
```
python3.7 -m pip install pre-commit --user
python3.7 -m pre_commit install
```
* [x] Create branch called `feature/pre-commit-integration`
* [x] Add [`.pre-commit-config.yaml`](http://pre-commit.com/) config, containing the following [hooks](http://pre-commit.com/hooks.html):
  * [x] no-commit-to-branch
  * [x] [add-trailing-comma](https://github.com/asottile/add-trailing-comma)
  * [x] [remove-tabs](https://github.com/Lucas-C/pre-commit-hooks)
  * [x] trailing-whitespace
  * [x] check-merge-conflict
  * [x] double-quote-string-fixer
  * [x] requirements-txt-fixer
  * [x] check-yaml
  * [x] check-docstring-first
  * [x] end-of-file-fixer
  * [x] name-tests-test
  * [x] check-ast
  * [x] debug-statements
  * [x] flake8
  * [x] [pydocstyle](https://github.com/chewse/pre-commit-mirrors-pydocstyle)
  * [x] `pylint` using "local" mode
  * [x] [mypy](https://github.com/pre-commit/mirrors-mypy.git)
* [x] Check repo using command `pre-commit run --all-files`, fix all issues and commit fixes as well
* [x] Create PR with this branch

### PR DESCRIPTION
**Pull request by RootenberG**
Closes #14

### CODE REVIEW

**Comment by webknjaz** [comment]
@RootenberG You need to put whitespace between a closing keyword and an issue number.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- repo: https://github.com/Lucas-C/pre-commit-hooks
+  sha: v1.1.6
+  hooks:
+  - id: remove-tabs
+- repo: git://github.com/chewse/pre-commit-mirrors-pydocstyle
+  sha: v2.1.1
+  hooks:
+  - id: pydocstyle
+    name: Verifying PEP257 Compliance
+    entry: pydocstyle
```
Comment (webknjaz) [inline .pre-commit-config.yaml]: Why do you need this?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+- repo: https://github.com/asottile/add-trailing-comma
+  rev: v1.3.0
+  hooks:
+  - id: add-trailing-comma
+- repo: https://github.com/Lucas-C/pre-commit-hooks
+  sha: v1.1.6
+  hooks:
+  - id: remove-tabs
+- repo: git://github.com/chewse/pre-commit-mirrors-pydocstyle
+  sha: v2.1.1
```
Comment (webknjaz) [inline .pre-commit-config.yaml]: `sha:` is outdated. you need to use `rev:` now

**Comment by igorkruglyak** [comment]
In my opinion, you need to change in pylint 
[https://github.com/kpi-web-guild/django-girls-blog-igorkruglyak/blob/5ee9d1896e1f8c78d1e5921938f930826e57895b/.pre-commit-config.yaml#L46-L53](url)

File: .pre-commit-config.yaml (hunk 1 above)
Comment (igorkruglyak) [reply .pre-commit-config.yaml]: I do not think it is need. These settings are already in
https://github.com/chewse/pre-commit-mirrors-pydocstyle/blob/master/.pre-commit-hooks.yaml
For an example you can just like here
https://github.com/kpi-web-guild/django-girls-blog-igorkruglyak/blob/5ee9d1896e1f8c78d1e5921938f930826e57895b/.pre-commit-config.yaml#L38-L39

**Comment by igorkruglyak** [comment]
> Closes #14

You can to write more comments. 
For example

Closes #14

Ref: https://pre-commit.com

This commit includes the initial configuration for
the following pre-commit hooks:

    no-commit-to-branch
    add-trailing-comma
    remove-tabs
    trailing-whitespace
    check-merge-conflict
    double-quote-string-fixer
    requirements-txt-fixer
    check-yaml
    check-docstring-first
    end-of-file-fixer
    name-tests-test
    check-ast
    debug-statements
    flake8
    pydocstyle
    pylint using "local" mode
    mypy

**Comment by igorkruglyak** [comment]
pylint you can write in this example:
https://github.com/kpi-web-guild/django-girls-blog-igorkruglyak/blob/5ee9d1896e1f8c78d1e5921938f930826e57895b/.pre-commit-config.yaml#L46-L53

**Review by DrEdi — COMMENTED** [review]
For me it’s ok, but first fix remarks from other reviewers.

Also remember, that some hooks could be in conflict (you’ll see it lately)

**Comment by webknjaz** [comment]
Plz clean up your commits

**Comment by RootenberG** [comment]
> Plz clean up your commits

What do you mean?

**Comment by webknjaz** [comment]
You have a fixup commit which by itself is only useful combined with another commit. Squash them into one atomic thing.

**Comment by RootenberG** [comment]
> You have a fixup commit which by itself is only useful combined with another commit. Squash them into one atomic thing.

Done.

**Review by webknjaz — APPROVED** [review]

**Review by DrEdi — APPROVED** [review]

**Comment by RootenberG** [comment]
I,m not authorized to merge this pull request.(

**Comment by webknjaz** [comment]
@RootenberG you can merge it now

**Pull request merged**